### DISCUSSION TITLE
Intel Alder/Raptor Lake P-cores and E-cores

### DISCUSSION THREAD

**Discussion by mlurbe97** [opcm/pcm#600] [Q&A] [answered]
I'm unable to program events on this platforms using the provided function
`pcm_c_build_core_event(event_id, event_name)`
inside a C code due to there are different events on each type of cores (P and E cores). 

Maybe a similar function should be implemented to program events per core
`pcm_c_build_core_event_on_core(cpu_id, event_id, event_name) `
or a warning should be shown.

Now I'm having 0s on E cores when configuring events for P cores, and viceversa. It could be interesting to monitor similar events on both.

**Comment by opcm** [comment, marked as answer]
to program P-cores and E-cores events separately you can try to use a different PCM::program interface with ExtendedCustomCoreEventDescription structure as a parameter: https://github.com/intel/pcm/blob/0cbcdbe3b9b3002b2db4d10a98fcb5b9f2dd4156/src/cpucounters.h#L859 

It has a separate Atom/E-core event config field: https://github.com/intel/pcm/blob/0cbcdbe3b9b3002b2db4d10a98fcb5b9f2dd4156/src/cpucounters.h#L864
